hapi-fees/src/main/java/com/hedera/services/usage/token/TokenMintUsage.java

```diff
@@ -41,9 +45,24 @@ TokenMintUsage self() {
 		return this;
 	}
 
+
+	public TokenMintUsage givenSubType(SubType subType) {
+		this.currentSubType = subType;
+		return this;
+	}
+
 	public FeeData get() {
+		var op = this.op.getTokenMint();
+
+		// TODO: Split logic for calculations
+		if (currentSubType == SubType.TOKEN_NON_FUNGIBLE_UNIQUE) {
+			var bytesToAdd = op.getMetadataList().stream().mapToInt(ByteString::size).sum();
+			usageEstimator.addBpt(bytesToAdd);
+			usageEstimator.addRbs(bytesToAdd);
```

Comment:
For each serial number we are adding not only the metadata, but `RichInstant` creation time, `serialNumber` and `owner`. You can see the GetNftInfo PR for the entity sizes, might be valuable